### PR TITLE
chore(deps): update cargo dependencies

### PR DESCRIPTION
**Pull request by mayeul-zama**
Update cargo dependencies

### CODE REVIEW

File: concrete-csprng/Cargo.toml [hunk 1]
```diff
@@ -19,8 +19,8 @@ libc = "0.2.133"
 
 [dev-dependencies]
 rand = "0.8.3"
-criterion = "0.3"
-clap = "=4.2.7"
+criterion = "0.5.1"
```
Comment (IceTDrinker) [inline concrete-csprng/Cargo.toml:22]: what about rayon above and libc ?

File: tasks/Cargo.toml [hunk 2]
```diff
@@ -6,7 +6,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-clap = "3.1"
+clap = "4.4.4"
```
Comment (IceTDrinker) [inline tasks/Cargo.toml:9]: please hardcode the version here, it's coming from core

File: concrete-csprng/Cargo.toml (hunk 1 above)
Comment (mayeul-zama) [reply concrete-csprng/Cargo.toml:22]: The last versions should be used as they are semver compatible with the specified ones

File: tasks/Cargo.toml (hunk 2 above)
Comment (mayeul-zama) [reply tasks/Cargo.toml]: Fixed

**Comment by mayeul-zama** [comment]
@slab-ci cpu_fast_test

**Review by IceTDrinker — APPROVED** [review]
Thanks ! Let's wait for the tests :)

**Review by tmontaigu — APPROVED** [review]

**Pull request merged**